modules/charset/escape.go

```diff
@@ -113,6 +116,8 @@ readingloop:
 				lineHasRTLScript = false
 				lineHasLTRScript = false
 
+			case runeCount == 1 && r == 0xFEFF: // UTF BOM
```

Comment:
Can we use the `i` variable instead for this?

```suggestion
			case i == 0 && r == 0xFEFF: // UTF BOM
```

Reply:
No, `i` is inside another loop.

Only `runeCount` is the correct counter for the whole content.